source/UF/SizeTruncationandConnectedness.lagda

```diff
@@ -0,0 +1,346 @@
+            (helper x x')
+            (f-is-surj x)
+            (f-is-surj x')
+\end{code}
+
+Lemma 2.3.
+
+\begin{code}
+
+ Lemma-2-3 : {X : 𝓤 ̇} {n : ℕ₋₂}
+           → Propositional-Resizing
+           → X is (n + 1) truncated
+           → X is ι (n + 2) locally-small
+ Lemma-2-3 {_} {X} {−2} pr X-prop =
+  pr X (is-prop'-implies-is-prop X-prop)
+ Lemma-2-3 {_} {_} {succ n} pr X-trunc x x' =
+  Lemma-2-3 pr (X-trunc x x')
+
+\end{code}
+
+Lemma 2.4.
+
+\begin{code}
+
+ Lemma-2-4 : {X : 𝓤 ̇} {Y : 𝓦 ̇} {f : X → Y} {n : ℕ₋₂}
+           → Propositional-Resizing
+           → f is (n + 1) truncated-map
+           → Y is ι (n + 2) locally-small
+           → X is ι (n + 2) locally-small
+ Lemma-2-4 {_} {_} {_} {_} {f} {_} pr f-trunc Y-loc-small =
+  locally-small-closed-under-≃ (total-fiber-is-domain f)
+   (locally-small-closed-under-Σ Y-loc-small (λ y → Lemma-2-3 pr (f-trunc y)))
+
+\end{code}
+
+Lemma 2.5.
+
+\begin{code}
+
+ Lemma-2-5 : {X : 𝓤 ̇} {Y : 𝓦 ̇} {f : X → Y} {n : ℕ₋₂}
+           → Join-Construction-Result {𝓤} {𝓤}
+           → Propositional-Resizing
+           → f is (n + 1) truncated-map
+           → Y is ι (n + 2) locally-small
+           → X is (n + 1) connected
+           → X is 𝓥 small
+ Lemma-2-5 {𝓤} {_} {X} {_} {_} {n} j pr f-trunc Y-loc-small X-conn =
+  ∥∥-rec (being-small-is-prop ua X 𝓥)
+   X-inhabited-implies-small (center X-−1-conn)
+  where
+   X-loc-small : X is ι (n + 2) locally-small
+   X-loc-small = Lemma-2-4 pr f-trunc Y-loc-small
+   X-−1-conn : X is −1 connected
+   X-−1-conn = connectedness-is-lower-closed' ⋆ X-conn
+   X-point : X → 𝟙 {𝓤} → X
+   X-point x ⋆ = x
+   X-point-n-conn : (x : X) → (X-point x) is n connected-map
+   X-point-n-conn x = basepoint-map-is-less-connected (ua 𝓤) (X-point x) X-conn
+   𝟙-is-small : 𝟙 {𝓤} is 𝓥 small
+   𝟙-is-small = pr 𝟙 𝟙-is-prop
+   X-inhabited-implies-small : X → X is 𝓥 small
+   X-inhabited-implies-small x =
+    Prop-2-2 j (X-point-n-conn x) 𝟙-is-small X-loc-small
+
+\end{code}
+
+We shall follow Dan's updated result and prove the following in the absence of
+propositional resizing.
+
+Theorem 2.6.
+
+\begin{code}
+
+ small-path-space-lemma : {X : 𝓤 ̇} {n : ℕ₋₂}
+                        → Join-Construction-Result {𝓤} {𝓤}
+                        → X is ι (n + 2) locally-small
+                         × ∥ X ∥[ n + 1 ] is 𝓥 small
+                        → (Σ y ꞉ ∥ X ∥[ n + 1 ] , Σ x ꞉ X , ∣ x ∣[ n + 1 ] ＝ y)
+                         is 𝓥 small
+ small-path-space-lemma {_} {X} {n} j (X-loc-small , trunc-X-small) =
+  Σ-is-small trunc-X-small fiber-path-space-small
+  where
+   trunc-ind-helper : (x' : X)
+                    → (Σ x ꞉ X , ∣ x ∣[ n + 1 ] ＝ ∣ x' ∣[ n + 1 ]) is 𝓥 small
+   trunc-ind-helper x' = Prop-2-2 j f-con 𝟙-is-small cod-locally-small
+    where
+     f : 𝟙 {𝓤} → Σ x ꞉ X , ∣ x ∣[ n + 1 ] ＝ ∣ x' ∣[ n + 1 ]
+     f ⋆ = (x' , refl)
+     cod-con : (Σ x ꞉ X , ∣ x ∣[ n + 1 ] ＝ ∣ x' ∣[ n + 1 ]) is (n + 1)
+                connected
```

Comment:
Maybe mode `is (n + 1)` to the next line. I think it would make it more easily readable.